flashinfer/autotuner/autotuner.py

```
@@ -1395,19 +1527,19 @@ def choose_one(
 
             return runners[runner_id], tactic
 
-    def _get_input_sizes(self, inputs: List[torch.Tensor]) -> List[torch.Size]:
+    def _get_input_sizes(
+        self, inputs: list[torch.Tensor]
+    ) -> tuple[tuple[int, ...], ...]:
         """Return ``torch.Size`` for each input, using ``(0,)`` for non-Tensor values."""
-        sizes = [
+        return tuple(
             input.size() if isinstance(input, torch.Tensor) else torch.Size((0,))
             for input in inputs
```

Ruff (0.15.18)

[error] Variable `input` is shadowing a Python builtin

(A001)